opentracing/src/main/java/opentracing/Span.java

```diff
@@ -62,29 +60,12 @@
   Span setTraceAttribute(String key, String value);
 
   /**
-   * {@code message} is a format string and can refer to fields in the payload by path, like so:
-   *
-   * <pre>{@code
-   *
-   * span.info("first transaction is worth ${transactions[0].amount} ${transactions[0].currency}",
-   *     ImmutableMap.of(
-   *       "transactions", asList(
-   *         Transaction.builder().amount(10).currency("USD").build(),
-   *         Transaction.builder().amount(11).currency("USD").build(),
-   *       )
-   *     )
-   * );
-   * }</pre>
-   *
-   * @param message {@link Formatter format string} that can refer to fields in the args payload.
-   * @param args arbitrary payload
-   */
-  // See https://github.com/opentracing/opentracing.github.io/issues/30 about parameterization
-  Span info(String message, Object... args);
-
-  /** Same as {@link #info}, but for warnings. */
-  Span warning(String message, Object... args);
+   * Add a timestamped annotation, with an optional payload.
+   **/
+  Span log(String message, /* @Nullable */ Object payload);
```

Comment:
Good question, and this may be an agree-to-disagree situation... If the debug-logging methods are named `info` and `error` (or just `info`, or also `warning`, or whatever), then there's no bona fide conflict. The benefit is just making it clear that `log()` is strictly the superset plumbing method and that the other special-purpose convenience methods have more descriptive names to distinguish them.

It sounds like you'd prefer to have the event-logging method just overload the plumbing API with a signature like `log(String, Object)`. My preference was to call it `logEvent(String, Object)` or even just `event(String, Object)`... that way `event` (or `logEvent`) is on equal footing with `info` and `error`, the other convenience methods.

Another non-possibility is to have all of the above overload `log`, but my original point was that this would be impossible since the event-logging and debug-logging signatures would look the same to the compiler.